src/app.rs

```diff
@@ -256,15 +267,19 @@ impl Application {
 
             let signal = loop {
                 tokio::select! {
-                    Some(signal) = signal_rx.recv() => {
+                    Ok(signal) = signal_rx.recv() => {
```

Comment:
Swapping signal handling to a broadcast channel changes this from an `Option<T>` to `Result<T>`.